src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const copyArray = [...this];
+
+    for (let i = 0; i < copyArray.length; i++) {
+      if (copyArray[i] === null || copyArray[i] === undefined) {
+        copyArray[i] = '';
+      }
```

Comment:
you're basically telling that if the char is null or undefined - `do nothing`
Let's try a different approach instead - add the char to the result if it is a valid string